infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationLogService.kt

```diff
@@ -664,99 +662,129 @@ constructor(
             }
         }
 
+    fun jointsString(jointNumbers: List<JointNumber>): String =
+        jointNumbers.joinToString("-") { j -> j.intValue.toString() }
+
     fun diffSwitch(
         translation: Translation,
         changes: SwitchChanges,
         newTimestamp: Instant,
         oldTimestamp: Instant,
-        operation: Operation,
         trackNumberCache: List<TrackNumberAndChangeTime>,
         geocodingContextGetter: (IntId<LayoutTrackNumber>, Instant) -> GeocodingContext<ReferenceLineM>?,
     ): List<PublicationChange<*>> {
-        val relatedJoints = changes.joints.filterNot { it.removed }.distinctBy { it.trackNumberId }
+        val getTrackNumber = { tnId: IntId<LayoutTrackNumber> ->
+            // If the track number name changed in the same publication, it has it's own change row
+            // -> just use the new name here for clarity, even for old address
+            requireNotNull(trackNumberCache.findLast { tn -> tn.id == tnId && tn.changeTime <= newTimestamp }?.number)
+        }
+        val getAddressChange = { pointChange: Change<Point?>, trackNumberId: IntId<LayoutTrackNumber> ->
+            val oldCtx = pointChange.old?.let { geocodingContextGetter(trackNumberId, oldTimestamp) }
+            val newCtx = pointChange.new?.let { geocodingContextGetter(trackNumberId, newTimestamp) }
+            val pointChanged =
+                when {
+                    pointChange.old == null && pointChange.new == null -> false
+                    pointChange.old == null || pointChange.new == null -> true
+                    else -> !pointChange.old.isSame(pointChange.new, LAYOUT_COORDINATE_DELTA)
+                }
+            if (oldCtx != newCtx || pointChanged) {
+                Change(
+                    old = pointChange.old?.let { p -> oldCtx?.getAddress(p)?.first },
+                    new = pointChange.new?.let { p -> newCtx?.getAddress(p)?.first },
+                )
+            } else null
+        }
 
-        val oldLinkedLocationTracks =
-            changes.locationTracks.associate { lt ->
-                locationTrackService.getWithGeometry(lt.oldVersion).let { (track, geometry) ->
```

Comment:
Tämän kutsun poistaminen oli myös osamotivaatio muuttaa näitä: jos tuo raide ei ole cachessa (kuten usein ei ole sillä kyseessä on vanhat versiot) niin joudutaan hakemaan raidegeometria kannasta. Koska uusi SQL palauttaa kaikki oleelliset sijaintitiedot (vähän liikaakin) tälle ei ole enää tarvetta.